pkg/liqoctl/install/kubeadm/provider.go

```diff
@@ -0,0 +1,87 @@
+package kubeadm
+
+import (
+	"context"
+	"fmt"
+
+	flag "github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	"github.com/liqotech/liqo/pkg/liqoctl/install/provider"
+)
+
+// NewKubeadmProviderConstructor initializes a new Kubeadm struct.
+func NewKubeadmProviderConstructor() provider.InstallProviderInterface {
```

Comment:
```suggestion
func NewProvider() provider.InstallProviderInterface {
```

Just a proposal though (kubeadm is already implied by the package name)